### PR TITLE
refactor(limits): Refactor global limits counter

### PR DESCRIPTION
**Pull request by zharinov**
## Changes:

Introduce `allowUnlimited` flag:

- `true` for commit per-run setting. Can be set  via CLI only.
- `false` for PR and branch per-repo limits. Options are set via config, but max limit values are calculated per-repo and can be 0 when limit reached.

## Context:

Ref: #7198

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added unit tests, or
- [ ] Unit tests + ran on a real repository

### CODE REVIEW

File: lib/workers/global/limits.ts [hunk 1]
```diff
@@ -15,15 +15,24 @@ export function resetAllLimits(): void {
...
+  val: unknown,
+  allowUnlimited = true
+): void {
+  if (typeof val === 'number') {
+    const current = 0;
+    let max: number = null;
+    if (allowUnlimited) {
+      if (val > 0) {
+        max = val;
+      }
```
Comment (viceice) [inline lib/workers/global/limits.ts:29]: Is this for backwar compability?

Comment (zharinov) [reply lib/workers/global/limits.ts:29]: Yes. Well, I think we can get rid of this flag completely.

File: lib/workers/global/limits.ts [hunk 2]
```diff
@@ -16,14 +16,13 @@ export function resetAllLimits(): void {
...
 export function setMaxLimit(key: Limit, max: unknown): void {
-  const maxVal = typeof max === 'number' && max > 0 ? max : null;
-  logger.debug(`${key} limit = ${maxVal}`);
-  const limit = limits.get(key);
-  limits.set(key, {
-    current: 0,
-    ...limit,
-    max: maxVal,
-  });
+  if (max === null || typeof max === 'number') {
```
Comment (rarkins) [inline lib/workers/global/limits.ts:19]: What's the difference here - `undefined` is skipped altogether?

Comment (zharinov) [reply lib/workers/global/limits.ts]: Never mind, `undefined` now also means limit reset

**Pull request merged**